src/toil/batchSystems/singleMachine.py

```diff
@@ -143,7 +143,7 @@ def _runWorker(self, jobCommand, jobID, environment):
         startTime = time.time()  # Time job is started
         popen = None
         statusCode = None
-        forkWorker = not self.debugWorker or "_toil_worker" not in jobCommand
+        forkWorker = not (self.debugWorker and "_toil_worker" not in jobCommand)
         if forkWorker:
```

Comment:
I thought it was threading only if "_toil_work" is in jobCommand. Did you want to fork for any non '_toil_worker' command, no matter the setting for self.debugWorker? If so, the second 'not' should be removed. Either way, a comment explaining the logic might be useful.

Reply:
The original behaviour was to always call `popen`, my hack should only be applied if and only if `_toil_worker` is present **and** `--debugWorker` is specified.

Did you see https://github.com/BD2KGenomics/toil/pull/1910#discussion_r142006836 ?

I'll add a comment with the above.

Reply:
The second 'not' needs to removed. Do you agree @mr-c? In parentheses, is the condition for threading, which is if self.debugWorker is True and "_toil_worker" is in jobCommand.